include/pathfinder.hpp

```diff
@@ -232,6 +261,10 @@ class pathfinder_base {
     //! lower the maximum chainlength
     int improve_chainlength_pass(embedding_t &emb) {
         bool improved = false;
+        dijkstras[0].reorder(params.rng);
+        for (int v = num_vars + num_fixed; v-- > 1;) {
```

Comment:
isn't this equivalent to the (standard): `for (int v = ...; v > 1; v--)`?

Reply:
also, is size of `dijkstras` >= `num_vars + num_fixed`? if `num_vars + num_fixed == 1`, should you skip the loop?

Reply:
1) yes, it looks like this works just as well?
2) i'm not going to waste a conditional at this point in the code to check if the overall problem is utterly trivial.  otoh... your suggested form for the loop apparently cuts out a postdecrement in the final iteration (I say "apparently" because modern compilers are too smart for that)

Reply:
ad 2) no, i wasn't suggesting you test for the trivial case; but depending on the answer to the first question (what's the size of `dijkstras` array), suggesting an error if you need to update `dijkstras[num_vars + num_fixed]`.

Reply:
`dijkstras` has size `num_vars+num_fixed`; it'd be an error to access that location ;)

Reply:
then ignore my second comment...